src/gui/Gui.cc

```diff
@@ -221,6 +221,16 @@ std::unique_ptr<gz::gui::Application> createGui(
     qputenv("QT_AUTO_SCREEN_SCALE_FACTOR", "1");
   }
 
+  // check for wayland and force to use X for rendering
+  if (QString::fromLocal8Bit(qgetenv("XDG_SESSION_TYPE")) == "wayland")
+  {
+    if (QString::fromLocal8Bit(qgetenv("QT_QPA_PLATFORM")).isEmpty())
+    {
```

Comment:
Perhaps log that we are automatically setting the variable to `XCB` here.

Reply:
added log msg here. [e2d8d33](https://github.com/gazebosim/gz-sim/pull/2526/commits/e2d8d332fd2d086d7dc079c02d58b4050740052c)